### PR TITLE
Make thread name mandatory in `run-thread'.

### PR DESCRIPTION
**Pull request by Ambrevar**
@jmercouris @pdelfino @aartaka Do you think it's a good idea?

### CODE REVIEW

File: libraries/prompter/prompter-source.lisp [hunk 1]
```diff
@@ -498,16 +498,12 @@ raised condition."
...
   "Run body in a new protected new thread.
 This is a \"safe\" wrapper around `bt:make-thread'."
-  ;; We parse the body instead of using a macro argument for backward compatibility.
-  (let* ((name (if (stringp (first body))
-                   (first body)
-                   "anonymous thread"))
-         (body (if (stringp (first body))
-                   (rest body)
-                   body)))
+  (let ((body (if (stringp (first body))
```
Comment (aartaka) [inline libraries/prompter/prompter-source.lisp:504]: But why? We already have `name`, then everything in `body` is, well... body of the function?

Am I missing something?

File: source/browser.lisp [hunk 2]
```diff
@@ -422,6 +422,7 @@ Return the download object matching the download."
                          (download-manager:filename download))
                    (push download-render (downloads *browser*))
                    (run-thread
+                     "Download watcher"
```
Comment (aartaka) [inline source/browser.lisp:425]: What's the convention for thread names, actually? I see that we append `nyxt` in front of those, so it should be lowercase, but other than that, it's a total chaos xD

File: source/concurrency.lisp [hunk 3]
```diff
@@ -70,17 +70,13 @@ This is a blocking operation."
...
 This supersedes `bt:make-thread' in Nyxt.  Don't use the latter unless you know
 what you are doing!"
-  ;; We parse the body instead of using a macro argument for backward compatibility.
-  (let* ((name (if (stringp (first body))
-                   (first body)
-                   "anonymous thread"))
-         (body (if (stringp (first body))
-                   (rest body)
-                   body)))
+  (let ((body (if (stringp (first body))
```
Comment (aartaka) [inline source/concurrency.lisp:77]: Same as in prompter-source, why parse body if `name` is already parsed for us?

**Review by aartaka — COMMENTED** [review]
I am confused by the `body` parsing and am not sure in understanding out thread naming conventions, but it's perfect otherwise :)

File: source/browser.lisp (hunk 2 above)
Comment (Ambrevar) [reply source/browser.lisp:425]: Hmmm, OK, we can put it all lower case.

**Comment by Ambrevar** [comment]
I believe I've addressed all your points!

**Comment by aartaka** [comment]
Yes, I see nothing problematic about this PR now :D

**Pull request merged**